File: client-scalaz/src/main/scala/org/openapitools/client/api/AssetResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import AssetResponse._

case class AssetResponse (
  valid: Option[Boolean],
message: Option[String],
version: Option[Double],
serializeNulls: Option[Boolean],
startTimeLog: Option[Long],
errorCode: Option[String],
request: Option[List[NameStringValueResponse]],
assetId: Option[Long],
fileName: Option[String],
caption: Option[String],
mediaType: Option[String],
thumbnailURL: Option[String],
coverURL: Option[String],
fullURL: Option[String],
assetVersion: Option[String],
versionName: Option[String],
versionCode: Option[Integer],
display: Option[String],
locationDescription: Option[String],
uploader: Option[AccountShortResponse],
likesCount: Option[Long],
commentsCount: Option[Integer],
latitude: Option[Double],
longitude: Option[Double],
attachedMediaType: Option[String],
attachedAssetURL: Option[String],
created: Option[Long],
updated: Option[Long],
searchTags: Option[String],
metaData: Option[String],
returning: Option[String])

object AssetResponse {
  import DateTimeCodecs._

  implicit val AssetResponseCodecJson: CodecJson[AssetResponse] = CodecJson.derive[AssetResponse]
  implicit val AssetResponseDecoder: EntityDecoder[AssetResponse] = jsonOf[AssetResponse]
  implicit val AssetResponseEncoder: EntityEncoder[AssetResponse] = jsonEncoderOf[AssetResponse]
}

File: client-clojure/src/sirqul_io_t_platform/specs/asset_response.clj
(ns sirqul-io-t-platform.specs.asset-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.name-string-value-response :refer :all]
            [sirqul-io-t-platform.specs.account-short-response :refer :all]
            )
  (:import (java.io File)))


(def asset-response-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :message) string?
   (ds/opt :version) float?
   (ds/opt :serializeNulls) boolean?
   (ds/opt :startTimeLog) int?
   (ds/opt :errorCode) string?
   (ds/opt :request) (s/coll-of name-string-value-response-spec)
   (ds/opt :assetId) int?
   (ds/opt :fileName) string?
   (ds/opt :caption) string?
   (ds/opt :mediaType) string?
   (ds/opt :thumbnailURL) string?
   (ds/opt :coverURL) string?
   (ds/opt :fullURL) string?
   (ds/opt :assetVersion) string?
   (ds/opt :versionName) string?
   (ds/opt :versionCode) int?
   (ds/opt :display) string?
   (ds/opt :locationDescription) string?
   (ds/opt :uploader) account-short-response-spec
   (ds/opt :likesCount) int?
   (ds/opt :commentsCount) int?
   (ds/opt :latitude) float?
   (ds/opt :longitude) float?
   (ds/opt :attachedMediaType) string?
   (ds/opt :attachedAssetURL) string?
   (ds/opt :created) int?
   (ds/opt :updated) int?
   (ds/opt :searchTags) string?
   (ds/opt :metaData) string?
   (ds/opt :returning) string?
   })

(def asset-response-spec
  (ds/spec
    {:name ::asset-response
     :spec asset-response-data}))
